src/agent/utils.ts

```diff
@@ -92,26 +89,35 @@ async function spawnBackgroundAgent( // FIXME, this is broken.
     externalReject = reject;
   });
 
-  agentProcess.send(JSON.stringify(startOptions), (err: Error) => {
+  // Const logProm = new Promise(async (res, _rej) => {
+  //   await sleep(30000);
+  //   console.log((fs.readFileSync(path.join(logPath, 'output.log'))).toString());
+  //   console.log((fs.readFileSync(path.join(logPath, 'error.log'))).toString());
+  //   await res(null);
+  // })
+  agentProcess.send(JSON.stringify(agentConfig), (err: Error) => {
     if (err != null) {
       agentProcess.kill('SIGTERM');
     } else {
       pid = agentProcess.pid;
       agentProcess.on('message', (msg) => {
         agentProcess.unref();
         agentProcess.disconnect();
-        if (msg !== 'started') {
+        const messageString = msg.toString().split(':');
+        if (messageString[0] !== 'started') {
           externalReject(
```

Comment:
Can you use exceptions here. It is not allowed to resolve to literal string. We have a linting rules for this. Also why not use try throw catch?

Reply:
As per the discussion this will be addressed in a separate issue.